src/test/java/org/assertj/core/api/Assertions_assertThat_asList.java

```diff
@@ -47,7 +48,7 @@ public void should_pass_list_asserts_on_list_strings_with_asList() {
   public void should_fail_list_asserts_on_non_list_objects_even_with_asList() {
     Object nonList = new Object();
 
-    thrown.expectAssertionError("an instance of:%n <java.util.List>%nbut was instance of:%n <java.lang.Object>");
+    thrown.expectAssertionErrorWithMessageContaining(format("an instance of:%n <java.util.List>%nbut was instance of:%n <java.lang.Object>"));
```

Comment:
not sure as you would not be able to use `String.format` parameters substitution as `expectAssertionErrorWithMessageContaining` already defines varargs, it could simply use `String.format` without params substitution to address the `%n` problem on different OS (if that what `format` usage was solving).
Long story short, I let you decide what is best ;-)